### PR TITLE
[SPARK-32162][PYTHON][TESTS] Improve error message of Pandas grouped map test with window

### PR DESCRIPTION
**Pull request by BryanCutler**
### What changes were proposed in this pull request?

Improve the error message in test GroupedMapInPandasTests.test_grouped_over_window_with_key to show the incorrect values.

### Why are the changes needed?

This test failure has come up often in Arrow testing because it tests a struct  with timestamp values through a Pandas UDF. The current error message is not helpful as it doesn't show the incorrect values, only that it failed. This change will instead raise an assertion error with the incorrect values on a failure.

Before:

```
======================================================================
FAIL: test_grouped_over_window_with_key (pyspark.sql.tests.test_pandas_grouped_map.GroupedMapInPandasTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/spark/python/pyspark/sql/tests/test_pandas_grouped_map.py", line 588, in test_grouped_over_window_with_key
    self.assertTrue(all([r[0] for r in result]))
AssertionError: False is not true
```

After:
```
======================================================================
ERROR: test_grouped_over_window_with_key (pyspark.sql.tests.test_pandas_grouped_map.GroupedMapInPandasTests)
----------------------------------------------------------------------
...
AssertionError: {'start': datetime.datetime(2018, 3, 20, 0, 0), 'end': datetime.datetime(2018, 3, 25, 0, 0)}, != {'start': datetime.datetime(2020, 3, 20, 0, 0), 'end': datetime.datetime(2020, 3, 25, 0, 0)}
```

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Improved existing test

### CODE REVIEW

**Comment by BryanCutler** [comment]
This is currently being looked at in https://github.com/apache/arrow/pull/7604

File: python/pyspark/sql/tests/test_pandas_grouped_map.py [hunk 1]
```diff
@@ -570,17 +570,21 @@ def test_grouped_over_window_with_key(self):
...
+
+            # Make sure the key with group and window values are correct
+            for _, i in pdf.id.iteritems():
+                assert expected[i][0] == group, "{} != {}".format(expected[i][0], group)
+                assert expected[i][1] == window_range, \
+                    "{} != {}".format(expected[i][1], window_range)
+
+            return pdf
 
         result = df.groupby('group', window('ts', '5 days')).apply(f).select('result').collect()
```
Comment (BryanCutler) [inline python/pyspark/sql/tests/test_pandas_grouped_map.py:589]: oops, I messed up removing the `result` col, I'll have to fix that up.

**Comment by BryanCutler** [comment]
ping @HyukjinKwon please take a look, thanks!

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by BryanCutler** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
I think .. the machines became very slow .. for some reasons.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Woah, finally

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Comment by BryanCutler** [comment]
Wow, Jenkins must have been slow back to work after the long weekend.. Thanks for the help @HyukjinKwon !

**Pull request closed without merging**